### PR TITLE
Changes from dmstr fork

### LINKED ISSUE #5: Plugin should require at least version 1.6.0 of yii2-usuario
It'll be fixed by PR#4

### PR DESCRIPTION
**Pull request by eluhr**
As already announced in Pull Request #2 , here are the changes from our development.

We added some changes to naming an the overall structure:

Module -> UsuarioLdapComponent: A module in Yii2 is used summarize models, views and controllers as well as other components. Naming it "Module" would be a bit misleading.

Moved initAdLdap method to getter and setter. So the ldap connection will only be opened when needed. Due to the bootstrapping of the component, that happened all the time.

Improved error handling and implemented some custom events.

CC: @schmunk42

### CODE REVIEW

**Comment by schmunk42** [comment]
We'd have some more improvements after this PR, but we wanted to keep it as minimal as possible regarding this PR.
Also we'd recommend to release a 3.0 as the next version since we already released a 2.x version in our fork.

**Comment by maxxer** [comment]
@mp1509 can you schedule a review? Thanks

**Comment by schmunk42** [comment]
@maxxer @mp1509 Ping :)

**Comment by maxxer** [comment]
Hi guys, thanks for pinging! @mp1509 has been working heavily on the extension and is going to push changes this week. We'll see what changes can be merged after his update.

Thanks again for the MR, I really hope we can join forces

File: README.md [hunk 1]
```diff
@@ -9,13 +9,13 @@ The preferred way to install this extension is through [composer](http://getcomp
 Either run:
 
 ```
-php composer.phar require --prefer-dist yetopen/yii2-usuario-ldap "*"
+php composer.phar require --prefer-dist yetopen/yii2-usuario-ldap-extension "*"
```
Comment (mp1509) [inline README.md:12]: I don't understand this change, in `composer.json` it is still named `yetopen/yii2-usuario-ldap`

Comment (schmunk42) [reply README.md:12]: Sorry, this is a leftover. Just remove it.

**Comment by mp1509** [comment]
Hi guys!
I've finally committed all the changes we had in store and tried to see if they could be easily merged with your branch. Unfortunately our last version is too different to merge yours automatically, so I had to do it manually.
I've pushed a new branch `eluhr-feature/marriage`, can you please verify if everything checks out for you in this new version?

I've also left a comment on this PR on a particular change.

Thank you!

**Comment by schmunk42** [comment]
@mp1509 Basically this, right?

https://github.com/YetOpen/yii2-usuario-ldap/compare/25a8e8d321860e2c95ed577e9796d3054a422dfd...eluhr-feature/marriage

CC: @eluhr

**Comment by schmunk42** [comment]
I tried `eluhr-feature/marriage` and just found these two issues:

Here's just a NS missing:
```
diff --git a/UsuarioLdapComponent.php b/UsuarioLdapComponent.php
index ab6d0ec..b6df823 100644
--- a/UsuarioLdapComponent.php
+++ b/UsuarioLdapComponent.php
@@ -361,7 +361,7 @@ class UsuarioLdapComponent extends Component
             $username_inserted = $username;
             $ldap_user = $this->findLdapUser($username, ['uid', 'cn', 'samaccountname', 'mail', 'userPrincipalName'], 'ldapProvider');
 
-            $username = ArrayHelper::getValue($ldap_user->getAttribute('uid'), '0');
+            $username = \yii\helpers\ArrayHelper::getValue($ldap_user->getAttribute('uid'), '0');
             if (empty($username)) {
                 $username = $username_inserted;
             }
```

I had to comment/remove the following lines:
```
@@ -435,8 +435,10 @@ class UsuarioLdapComponent extends Component
             // Now I have a valid user which passed LDAP authentication, we remove any error that may stop the login based on previous local authentication
             $form->clearErrors('password');
             $clIdentityUser = $this->make(User::class);
-            $userIdentity = $clIdentityUser::findIdentity($user->id);
-            $form->setUser($userIdentity);
+
+            #TODO: form has no setUser(?), lines can be removed --- login still works
+            #$userIdentity = $clIdentityUser::findIdentity($user->id);
+            #$form->setUser($userIdentity);
 
             Yii::info("The user '{$user->username}' has successfully logged in via LDAP", "ACCESSO_LDAP");
         });
```

CC: @eluhr

**Comment by schmunk42** [comment]
I can not push onto this repo.

```
ERROR: Permission to YetOpen/yii2-usuario-ldap.git denied to schmunk42.
```

So I've pushed it here: https://github.com/schmunk42/yii2-usuario-ldap/tree/eluhr-feature/marriage

**Comment by mp1509** [comment]
Thank you! I've left a couple of comments on the last commit of your branch

**Comment by schmunk42** [comment]
updated PR https://github.com/YetOpen/yii2-usuario-ldap/pull/6

**Comment by mp1509** [comment]
@schmunk42 I've merged your branch, I'll update the CHANGELOG and make a new release asap, but for now you should be aligned with our `master` repo.

**Comment by schmunk42** [comment]
@mp1509 Nice.

Could you release a `3.x` since we have already used `2.x`.

**Pull request merged**